content/library/advanced-features/dataframes.md

```diff
@@ -228,298 +224,88 @@ st.data_editor({
 
-Whether it's translating temporal data into user-friendly formats or utilizing charts and progress bars for clearer data visualization, column configuration not only provides the user with an enriched data viewing experience but also ensures that you're equipped with the tools to present and interact with your data, just the way you want it.
+Column configuration includes Text, Number, Checkbox, Selectbox, Date, Time, Datetime, List, Link, Image, Line chart, Bar chart, and Progress bar columns. There is also a generic Column option. See the embedded app below to view these different column types. Each column type is individually previewed in the [Column configuration API](/library/api-reference/data/st.column_config) documentation.
 
-<TileContainer>
-<RefCard href="/library/api-reference/data/st.column_config/st.column_config.column">
-<Image pure alt="screenshot" src="/images/api/column_config.column.jpg" />
+<Cloud src="https://doc-column-config-overview.streamlit.app/?embed=true&embed_options=disable_scrolling" height="480"/>
 
-#### Column
+### Format values
 
-Configure a generic column.
+You can specify the format of your values in two ways: through column configuration or through [`pandas.DataFrame.style`](https://pandas.pydata.org/docs/reference/style.html). When using `st.data_editor`, formatting for editable columns must be declared through column configuration and not a pandas `Styler` object. Only disabled columns in a data editor can accept formatting from a pandas `Styler` object.
 
-```python
-Column("Streamlit Widgets", width="medium", help="Streamlit **widget** commands 🎈")
-```
+A `format` parameter is available in column configuration for Text, Date, Time, and Datetime columns. Chart-like columns can also be formatted. Line chart and Bar chart columns have a `y_min` and `y_max` parameters to set the vertical bounds. You can declare the horizontal bounds with `min_value` and `max_value` for a Progress column.
 
-</RefCard>
-<RefCard href="/library/api-reference/data/st.column_config/st.column_config.textcolumn">
-<Image pure alt="screenshot" src="/images/api/column_config.textcolumn.jpg" />
+### Validate input
 
-#### Text column
+When specifying a column configuration, you can declare not only the data type of the column but also value restrictions. All column configuration elements allow you to make a column required with the keyword parameter `required=True`.
 
-Configure a text column.
-
-```python
-TextColumn("Widgets", max_chars=50, validate="^st\.[a-z_]+$")
-```
-
-</RefCard>
-
-<RefCard href="/library/api-reference/data/st.column_config/st.column_config.numbercolumn">
-<Image pure alt="screenshot" src="/images/api/column_config.numbercolumn.jpg" />
+#### Text and Link columns
```

Comment:
I'm OK with listing them out explicitly as here, but I could also imagine making it a bit shorter and just saying something like "e.g. for text columns, you can set a maximum number of characters through `max_chars` or use regex validation in `validate`". Feels a bit like we are just repeating the API docs here...

Reply:
Concise for the win!